lib/src/Uncertainty/Bayesian/GaussianLinearCalibration.cxx

```diff
@@ -113,33 +113,54 @@ GaussianLinearCalibration::GaussianLinearCalibration(const Sample & modelObserva
 /* Performs the actual computation. Must be overloaded by the actual calibration algorithm */
 void GaussianLinearCalibration::run()
 {
+  // Compute the difference of output observations and output predictions
   const Point deltaY(modelObservations_.getImplementation()->getData() - outputObservations_.getImplementation()->getData());
+  // Compute covariance matrix B of the parameter and its Cholesky decomposition
   CovarianceMatrix B(getParameterCovariance());
-  const IdentityMatrix IB(B.getDimension());
-  const CovarianceMatrix invB(B.solveLinearSystem(IB).getImplementation());
+  const TriangularMatrix LB(B.computeCholesky());
+  // Compute the covariance matrix R
   CovarianceMatrix R(deltaY.getSize());
   const UnsignedInteger dimension = errorCovariance_.getDimension();
+  const UnsignedInteger size = outputObservations_.getSize();
   if (globalErrorCovariance_) R = errorCovariance_;
   else
     {
       if (dimension == 1) R = (R * errorCovariance_(0, 0)).getImplementation();
       else
 	{
-	  const UnsignedInteger size = outputObservations_.getSize();
 	  for (UnsignedInteger i = 0; i < size; ++i)
 	    for (UnsignedInteger j = 0; j < dimension; ++j)
 	      for (UnsignedInteger k = 0; k < dimension; ++k)
 		R(i * dimension + j, i * dimension + k) = errorCovariance_(j, k);
 	}
     }
-  const IdentityMatrix IR(R.getDimension());
-  const CovarianceMatrix invR(R.solveLinearSystem(IR).getImplementation());
```

Comment:
Same thing here: first, create the error distribution, then extract the relevant matrices.